pkg/claims/claims.go

```diff
@@ -43,5 +54,22 @@ func OwnedByVMLabel(vmiName string) client.MatchingLabels {
 }
 
 func ComposeKey(vmName, networkName string) string {
-	return fmt.Sprintf("%s.%s", vmName, networkName)
+	return convertToCRDName(vmName + claimKeySeparator + networkName)
+}
+
+func convertToCRDName(input string) string {
+	// Convert to lowercase
+	crdName := strings.ToLower(input)
+
+	// Find the longest valid substring that matches the pattern
+	matches := rfc1123SubdomainsRegexp.FindAllString(crdName, -1)
+
+	// Join valid substrings with a hyphen (if there are multiple matches)
+	crdName = strings.Join(matches, claimKeySeparator)
+
+	// Truncate to 253 characters if necessary
+	if len(crdName) > 253 {
```

Comment:
maybe turn 253 into a const ?

Reply:
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names
just took about RFC 1123, in case the rules fit here as well
```
RFC 1123 Label Names 
Some resource types require their names to follow the DNS label standard as defined in [RFC 1123](https://tools.ietf.org/html/rfc1123). This means the name must:

contain at most 63 characters
contain only lowercase alphanumeric characters or '-'
start with an alphanumeric character
end with an alphanumeric character
```

aint it should be 63 ?
(maybe any other rule interesting here)

Reply:
tks for actually reading through the standard Or.

I think @oshoval 's right.

Reply:
I see 253 here -> https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names

Reply:
hm, so:
- DNS Subdomain Names: max len => 253
- RFC 1123 Label Names: max len => 63

Seems we got to figure out exactly which we're handling. I think we follow DNS subdomain names - from the [custom resources docs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions):
>  the name of an object whose kind/resource is defined by a CRD must also be a valid DNS subdomain name

Let's keep 253. Or ... better yet. Check it out. I'm nearly sure you can test this with the fakeclient structs.

Reply:
We have to follow what kubernetes mandates, since the implementation of kubernetes is doing so.